include/types.h

```diff
@@ -147,11 +147,38 @@ struct range {
     bool within(const range& r) const noexcept { return rid == r.rid && beg >= r.beg && end <= r.end; }
     bool contains(const range& r) const noexcept { return r.within(*this); }
 
+    bool spanned_by(std::vector<range>& record_rngs) const noexcept {
```

Comment:
Not a big deal but I find the implementation of `spanned_by` is a little bit convoluted, calling a private member function that calls back out to other members...I wonder can we think of ways to simplify. (Not urgent, can file as a future clean-up)